packages/widgets/src/_inputs/widget-app-input.tsx

```diff
@@ -18,42 +20,63 @@ export const WidgetAppInput = ({ property, kind }: CommonWidgetInputProps<"app">
   const t = useI18n();
   const tInput = useWidgetInputTranslation(kind, property);
   const form = useFormContext();
-  const { data: apps, isPending } = clientApi.app.selectable.useQuery();
+  const { data: apps, isPending, refetch } = clientApi.app.selectable.useQuery();
+
+  const { openModal } = useModalAction(QuickAddAppModal);
 
   const currentApp = useMemo(
     () => apps?.find((app) => app.id === form.values.options.appId),
     [apps, form.values.options.appId],
   );
 
   return (
-    <Select
-      label={tInput("label")}
-      searchable
-      limit={10}
-      leftSection={<MemoizedLeftSection isPending={isPending} currentApp={currentApp} />}
-      nothingFoundMessage={t("widget.common.app.noData")}
-      renderOption={renderSelectOption}
-      data={
-        apps?.map((app) => ({
-          label: app.name,
-          value: app.id,
-          iconUrl: app.iconUrl,
-        })) ?? []
-      }
-      inputWrapperOrder={["label", "input", "description", "error"]}
-      description={
-        <Text size="xs">
-          {t.rich("widget.common.app.description", {
-            here: () => (
-              <Anchor size="xs" component={Link} target="_blank" href="/manage/apps/new">
-                {t("common.here")}
-              </Anchor>
-            ),
-          })}
-        </Text>
-      }
-      {...form.getInputProps(`options.${property}`)}
-    />
+    <Group>
+      <Select
+        label={tInput("label")}
+        searchable
+        limit={10}
+        leftSection={<MemoizedLeftSection isPending={isPending} currentApp={currentApp} />}
+        nothingFoundMessage={t("widget.common.app.noData")}
+        renderOption={renderSelectOption}
+        data={
+          apps?.map((app) => ({
+            label: app.name,
+            value: app.id,
+            iconUrl: app.iconUrl,
+          })) ?? []
+        }
+        inputWrapperOrder={["label", "input", "description", "error"]}
+        description={
+          <Text size="xs">
+            {t.rich("widget.common.app.description", {
+              here: () => (
+                <Anchor size="xs" component={Link} target="_blank" href="/manage/apps/new">
+                  {t("common.here")}
+                </Anchor>
+              ),
+            })}
+          </Text>
+        }
+        styles={{ root: { flex: "1" } }}
+        {...form.getInputProps(`options.${property}`)}
+      />
+      <Button
+        mt={3}
+        rightSection={<IconRocket size="1.5rem" />}
+        variant="default"
+        onClick={() =>
+          openModal({
+            // eslint-disable-next-line no-restricted-syntax
+            async onClose(createdAppId) {
+              await refetch();
+              form.setFieldValue(`options.${property}`, createdAppId);
+            },
+          })
+        }
+      >
+        {t("widget.common.app.quickCreate")}
+      </Button>
```

Comment:
To also look good on mobile devices I would suggest an `ActionIcon.Group` with a `ActionIcon.GroupSection` for the input

Reply:
Done:
![image](https://github.com/user-attachments/assets/9154f98d-7cc6-4d36-9020-730b5c93afcf)
![image](https://github.com/user-attachments/assets/39fc02c0-a8d5-448f-8cf0-582f4ca78fa7)

The ActionIcon didn't look good and didn't work - hence I used this solution.

Reply:
I don't like that it takes so much attention to this button, but yeah let's leave it like that for now